lint/libs/lint-checks/src/main/java/com/android/tools/lint/checks/CutPasteDetector.java

```diff
@@ -165,137 +168,201 @@ public void visitMethod(@NonNull JavaContext context, @Nullable UastVisitor visi
 
     @Nullable
     private static String getLhs(@NonNull UCallExpression call) {
-        UElement parent = skipParentheses(call.getContainingElement());
-        if (parent != null && UastExpressionUtils.isTypeCast(parent)) {
-            parent = parent.getContainingElement();
-        }
-
-        if (parent instanceof ULocalVariable) {
-            return ((ULocalVariable) parent).getName();
-        } else if (parent instanceof UBinaryExpression) {
-            UBinaryExpression be = (UBinaryExpression) parent;
-            UExpression left = be.getLeftOperand();
-            if (left instanceof UReferenceExpression) {
-                return left.asRenderString();
-            } else if (left instanceof UArrayAccessExpression) {
-                UArrayAccessExpression aa = (UArrayAccessExpression) left;
-                return aa.getReceiver().asSourceString();
-            }
-        } else if (UastExpressionUtils.isAssignment(parent)) {
-            //noinspection ConstantConditions
-            UExpression left = ((UBinaryExpression) parent).getLeftOperand();
-            if (left instanceof UReferenceExpression) {
-                return left.asSourceString();
-            } else if (left instanceof UArrayAccessExpression) {
-                UArrayAccessExpression aa = (UArrayAccessExpression) left;
-                return aa.getReceiver().asSourceString();
+        UElement parent = call.getContainingElement();
+        while (parent != null && !(parent instanceof UBlockExpression)) {
```

Comment:
Looks like we should stop also on any declaration (`!(parent instanceof UDeclaration`).